### PR TITLE
[JW8-1838] FloatOnScroll with wrapper

### PR DESCRIPTION
**Pull request by pajong**
### Are there any points in the code the reviewer needs to double check?
Current issues:
- Placeholder for percentage width and aspect ratio has height 0.
- Header label (and possibly dismiss button) do not float properly.

#### Addresses Issue(s):

JW8-1838

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  🗿 Set a milestone. It should be the ticket's fix version in JIRA.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/view/utils/views-manager.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const isAndroidChrome = OS.android && Browser.chrome;
 let intersectionObserver;
 let responsiveRepaintRequestId = -1;
 
-function lazyInitIntersectionObserver() {
+function lazyInitIntersectionObserver(threshold) {
```
Comment (vseventer) [inline src/js/view/utils/views-manager.js]: Don't think we need this any more.

File: src/js/view/utils/views-manager.js [hunk 2]
```diff
@@ -23,14 +23,14 @@ function lazyInitIntersectionObserver() {
...
-        }, { threshold: [0, 0.25, 0.5, 0.75, 1] });
+        }, threshold || { threshold: [0, 0.25, 0.5, 0.75, 1] });
     }
 }
 
 function matchIntersection(entry, group) {
     for (let i = group.length; i--;) {
         const view = group[i];
-        if (entry.target === view.getContainer()) {
+        if (entry.target === view.getContainer() || entry.target === view.getWrapper()) {
```
Comment (vseventer) [inline src/js/view/utils/views-manager.js]: Do we need this?

File: src/css/jwplayer/flags/floatingplayer.less [hunk 3]
```diff
@@ -0,0 +1,44 @@
+.jw-flag-floating {
+    background-size: cover;
+    overflow: visible;
+
+    .jw-wrapper {
+        overflow: visible;
+        animation: jw-float 150ms cubic-bezier(0, 0.25, 0.25, 1) forwards 1;
+        bottom: 1rem;
+        right: 1rem;
+        z-index: 1;
```
Comment (robwalch) [inline src/css/jwplayer/flags/floatingplayer.less]: What's this z-index for @radium-v @vseventer ?

**Comment by vseventer** [comment]
@robwalch @pajong I see you already removed the `z-index`, but in my opinion we should set it to at least `1`. Already in the harness, some elements (for example, the event count boxes) have a `z-index` of 1, and will be on top of the floating player.

I realize there might always be some elements with a higher `z-index`, so if we want to remove all-together and take note of this, that's fine with me too.

**Comment by vseventer** [comment]
Added outstanding issues in PR opening post.

**Comment by pajong** [comment]
Verified locally that the failing tests are not due to the changes from this PR.

**Pull request merged**